src/log-store/src/kafka/consumer.rs

```diff
@@ -0,0 +1,375 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::VecDeque;
+use std::ops::Range;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+
+use common_telemetry::debug;
+use futures::future::{BoxFuture, Fuse, FusedFuture};
+use futures::{FutureExt, Stream};
+use pin_project::pin_project;
+use rskafka::client::partition::PartitionClient;
+use rskafka::record::RecordAndOffset;
+
+use super::index::{NextBatchHit, RegionWalIndexIterator};
+
+#[async_trait::async_trait]
+pub trait FetchClient: std::fmt::Debug + Send + Sync {
+    /// Fetch records.
+    ///
+    /// Arguments are identical to [`PartitionClient::fetch_records`].
+    async fn fetch_records(
+        &self,
+        offset: i64,
+        bytes: Range<i32>,
+        max_wait_ms: i32,
+    ) -> rskafka::client::error::Result<(Vec<RecordAndOffset>, i64)>;
+}
+
+#[async_trait::async_trait]
+impl FetchClient for PartitionClient {
+    async fn fetch_records(
+        &self,
+        offset: i64,
+        bytes: Range<i32>,
+        max_wait_ms: i32,
+    ) -> rskafka::client::error::Result<(Vec<RecordAndOffset>, i64)> {
+        self.fetch_records(offset, bytes, max_wait_ms).await
+    }
+}
+
+struct FetchResult {
+    records_and_offsets: Vec<RecordAndOffset>,
+    batch_size: usize,
+    fetch_bytes: i32,
+    watermark: i64,
+    used_offset: i64,
+}
+
+#[pin_project]
+pub struct Consumer {
```

Comment:
Please add some docs for those structs, fields and methods.

Reply:
Added